appsec/src/extension/ddappsec.c

```diff
@@ -438,31 +343,29 @@ __thread void *unspecnull TSRMLS_CACHE = NULL;
 
 static void _check_enabled()
 {
-    if (!get_global_DD_APPSEC_TESTING() &&
-        (!dd_trace_enabled() || strcmp(sapi_module.name, "cli") == 0 ||
-            (sapi_module.phpinfo_as_text != 0))) {
-        DDAPPSEC_G(enabled_by_configuration) = DISABLED;
-    } else if (!dd_is_config_using_default(DDAPPSEC_CONFIG_DD_APPSEC_ENABLED)) {
-        DDAPPSEC_G(enabled_by_configuration) =
-            get_DD_APPSEC_ENABLED() ? ENABLED : DISABLED;
+    if ((!get_global_DD_APPSEC_TESTING() && !dd_trace_enabled()) ||
+        (strcmp(sapi_module.name, "cli") != 0 && sapi_module.phpinfo_as_text)) {
+        DDAPPSEC_G(enabled) = APPSEC_FULLY_DISABLED;
+        DDAPPSEC_G(active) = false;
+        DDAPPSEC_G(to_be_configured) = false;
+    } else if (!dd_cfg_enable_via_remcfg()) {
+        DDAPPSEC_G(enabled) = get_DD_APPSEC_ENABLED() ? APPSEC_FULLY_ENABLED
+                                                      : APPSEC_FULLY_DISABLED;
+        DDAPPSEC_G(active) = get_DD_APPSEC_ENABLED() ? true : false;
+        DDAPPSEC_G(to_be_configured) = false;
```

Comment:
Not sure If I missed something but, when appsec is enabled dynamically by remote config, it status can change many times. So saying `it does not have to be configure` sounds confusing.

Reply:
"To be configured" means "yet to be configured" (=not configured yet") or "it is to be configured" (=it's arranged/programmed for it to be configured in the future"). The purpose of this setting is just to indicate in MINIT if we got a configuration setting for the state of remote config yet (if, for instance, we can't contact the daemon, then to_be_configured will stay with the value `true`). In this branch, appsec is explicitly enabled or disabled, so we're never waiting for a setting from remote config. Does this make sense? I also added some comments in bbdb00390